### PR TITLE
CORDA-599 PersistentNetworkMapCache no longer circularly depends on SH

### PR DESCRIPTION
**Pull request by andr3ej**

### CODE REVIEW

File: core/src/main/kotlin/net/corda/core/node/services/NetworkMapCache.kt [hunk 1]
```diff
@@ -47,19 +62,6 @@ interface NetworkMapCache {
     fun track(): DataFeed<List<NodeInfo>, MapChange>
 
     /**
-     * Look up the node info for a specific party. Will attempt to de-anonymise the party if applicable; if the party
```
Comment (mnesbit) [inline core/src/main/kotlin/net/corda/core/node/services/NetworkMapCache.kt]: This is an API change and we are not allowed to make those

File: node/src/main/kotlin/net/corda/node/services/network/PersistentNetworkMapCache.kt [hunk 2]
```diff
@@ -41,15 +40,32 @@ import java.security.SignatureException
 import java.util.*
 import javax.annotation.concurrent.ThreadSafe
 
+class NodeLookupImpl(private val identityService: IdentityService, private val networkMapCache: NetworkMapCache) : NodeLookup {
```
Comment (mnesbit) [inline node/src/main/kotlin/net/corda/node/services/network/PersistentNetworkMapCache.kt]: I like the idea, but perhaps we have to keep it hidden due to api effects.

**Comment by rick-r3** [comment]
This looks like it changes the public API.

**Comment by rick-r3** [comment]
I think we should add @shamsasari since the network map is his area.

**Comment by andr3ej** [comment]
OK, then I think the best I can do here is retain the circularity but make it more explicit, like in the initial impl of #1630

**Comment by Clintonio** [comment]
I'd like this to be updated or closed with a rebase against master first to double check if the new API stability versions allow this or not.

**Comment by andr3ej** [comment]
I have a remix that works, and can be pushed tomorrow after some polishing. Keeping this PR open to highlight the difference between a change without API stability and the same change with.

**Comment by andr3ej** [comment]
Cool, the remaining api changes aren't real, they're 16 methods moved from NMC to NMCLite that NMC extends.

File: core/src/main/kotlin/net/corda/core/node/services/NetworkMapCache.kt [hunk 3]
```diff
@@ -29,6 +28,22 @@ interface NetworkMapCache {
...
+     *
+     * @param party party to retrieve node information for.
+     * @return the node for the identity, or null if the node could not be found. This does not necessarily mean there is
+     * no node for the party, only that this cache is unaware of it.
+     */
+    fun getNodeByLegalIdentity(party: AbstractParty): NodeInfo?
+}
+
+/** Subset of [NetworkMapCache] that doesn't depend on an [IdentityService]. */
+interface NetworkMapCacheLite {
```
Comment (Clintonio) [inline core/src/main/kotlin/net/corda/core/node/services/NetworkMapCache.kt]: This name is a bit weird. Maybe "NetworkMapCacheBase"?

File: node/src/main/kotlin/net/corda/node/internal/AbstractNode.kt [hunk 4]
```diff
@@ -459,8 +459,12 @@ abstract class AbstractNode(config: NodeConfiguration,
         attachments = NodeAttachmentService(services.monitoringService.metrics)
         cordappProvider.start(attachments)
         legalIdentity = obtainIdentity(notaryConfig = null)
+        // TODO  We keep only notary identity as additional legalIdentity if we run it on a node . Multiple identities need more design thinking.
```
Comment (Clintonio) [inline node/src/main/kotlin/net/corda/node/internal/AbstractNode.kt]: This function is getting pretty long but I see why you're doing this (so ordering is explicit and DI)

File: core/src/main/kotlin/net/corda/core/node/services/NetworkMapCache.kt (hunk 3 above)
Comment (andr3ej) [reply core/src/main/kotlin/net/corda/core/node/services/NetworkMapCache.kt]: i think guava uses similar naming. anyway it's not important, renamed

File: node/src/main/kotlin/net/corda/node/internal/AbstractNode.kt (hunk 4 above)
Comment (andr3ej) [reply node/src/main/kotlin/net/corda/node/internal/AbstractNode.kt]: this function will be a victim of CORDA-599

**Pull request merged**